### PR TITLE
feat: first step of adding scn

### PR DESCRIPTION
**Pull request by andrewKapustin22**

### CODE REVIEW

**Review by aleksgrin — APPROVED** [review]

**Review by ACherkas12 — APPROVED** [review]

File: src/components/Scn/components/ScLink/ScLink.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+//     return {
+//       ...acc,
+//       [curr.name]: curr.value,
+//     };
+//   }, {});
+// };
+
+const HtmlNodeToReact = ({ node }: IHtmlNodeToReactProps) => {
+  if (node instanceof Text) return <>{node.textContent}</>;
+  if (node.nodeName === 'SC_ELEMENT') {
```
Comment (Kovallevz) [inline src/components/Scn/components/ScLink/ScLink.tsx:37]: Do you form the `nodeName` yourself or do you define it beforehand, for example, in types?

File: src/components/Scn/components/ScLink/ScLink.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import { useClient } from '@components/ClientProvider';
+import { TLinkFormat } from '@components/Scn';
+import { useInView } from '@hooks/useInView';
+import { ScAddr } from 'ts-sc-client';
+
+import { voidElements } from './constants';
+import { StyledScnLink, StyledScTag, StyledScTagLink } from './styled';
+import { useDifferedLinkContent } from './useDifferedLinkContent';
+
+interface IProps {
```
Comment (Kovallevz) [inline src/components/Scn/components/ScLink/ScLink.tsx:11]: Why not change the name to something more informative here, too?

File: src/components/Scn/components/Nodes/Nodes.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <StyledScnLink addr={addr} />
+      <KeywordLinkWrapper>
+        <LeftSide>=</LeftSide>
+        <LinkNode tree={tree} />
+      </KeywordLinkWrapper>
+    </>
+  );
+};
+
+// export const KeywordNode: FC<INodeProps> = ({ children, tree }) => {
```
Comment (ACherkas12) [inline src/components/Scn/components/Nodes/Nodes.tsx:28]: There is an unnecessary comment

File: src/components/Scn/components/ScLink/ScLink.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  addr: number;
+  contentType?: TLinkFormat | null;
+  content?: string | null;
+}
+
+interface IHtmlNodeToReactProps {
+  node: ChildNode;
+}
+
+// const nodeAttributesToObject = (node: HTMLElement) => {
```
Comment (ACherkas12) [inline src/components/Scn/components/ScLink/ScLink.tsx:21]: Is it unnecessary comment?

File: src/components/Scn/Scn.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import { ScnElement } from './components/ScnElement';
+import { ScnSkeleton } from './components/ScnSkeleton';
+import { IScnNode } from './model';
+import { ScnProvider, TOnAskQuestion } from './ScnContext';
+import { Inner, StyledScTag, Target } from './styled';
+
+const PAGE_SIZE = 25;
+
+interface IProps {
```
[the remark this answers is not in the document]
Comment (andrewKapustin22) [reply src/components/Scn/Scn.tsx:12]: In case, when interface doesn't export, the name of props interface should be IProps

**Pull request merged**